packages/@aws-cdk/core/lib/token.ts

```diff
@@ -129,6 +130,22 @@ export class Tokenization {
     return isResolvableObject(obj);
   }
 
+  /**
+   * Stringify a number directly or lazily if it's a Token. If it is an object (i.e., { Ref: 'SomeLogicalId' }), return it as-is.
+   */
+  public static stringifyNumber(x: number) {
+    // only convert non-objects to strings so that Refs, conditions, and other things don't end up synthesizing as [object object]
+
+    if (Token.isUnresolved(x)) {
+      return Lazy.stringValue({ produce: context => {
+          const resolved = context.resolve(x);
+          return typeof resolved === 'object' ? resolved : `${resolved}`;
+        } });
+    } else {
+      return typeof x === 'object' ? x : `${x}`;
```

Comment:
This line can't be necessary. Do you know why this is here?

Reply:
if the input isn't tokenized it gets called

i.e.,:

```
    const number = new CfnParameter(this, 'Number', {
      type: 'Number',
    })
    const taskDefinition = new FargateTaskDefinition(this, 'TaskDefinition', {
      memoryLimitMiB: new Intrinsic({ Ref: 'Number' }),
    })
```

or

```
    const taskDefinition = new FargateTaskDefinition(this, 'TaskDefinition', {
      memoryLimitMiB: 123
    })
```

Reply:
really it should only stringify numbers - would it be clearer to change to typeof !== number?

This would have the added advantage that if you were to pass true/false, the validation of CfnTaskDefinitionProps would fail rather than converting it to the string 'true'/'false', for example.

Reply:
Yes, I think that would be good. Also, for example this:

```
      memoryLimitMiB: new Intrinsic({ Ref: 'Number' }),
```

Shouldn't type check in the first place, so people shouldn't be able to do this. They should be calling `Token.asNumber()` on this first.

Reply:
I've updated my PR... Oh I haven't even used the Intrinsic class ;) The issue is just with parameters for me, just an example ;)

I've updated to just convert numbers... Was there a change you'd like to disallow direct Intrinsic usage?

(sorry for the delay in updating! Been busy!)